### ISSUE TITLE
Feature Request: API to translate map coords to screen coords

### ISSUE THREAD

**Issue by chrisburnor** [StarArawn/bevy_ecs_tilemap#30]
For action games, the sprite entities would be 'free-floating' entities rendered on top of the map, but the game would still want to track their position relative to the map itself.  It probably doesn't make sense to have the map *store* those entities or references to them, but it would be useful to have the map translate a position (given as a `DVec2`) and return a `Transform` or `DVec2` object that corresponds to the screen coordinates of that location given the map projection and scale and vice versa.

**Comment by StarArawn** [comment]
Yes, this would be useful. I can provide an API like:
```rust
let map_pos: UVec2 = map_query.pixel_pos_to_map_pos(Vec2::new(0.0, 0.0));
assert!(map_pos == UVec2::new(0, 0));

// Assuming tiles are 16x16.
let pixel_pos: Vec2 = map_query.map_pos_to_pixel_pos(UVec2::new(0, 0), true); // boolean represents centered or not?
assert!(map_pos == Vec2::new(8.0, 8.0)); 
```

**Comment by chrisburnor** [comment]
That looks good! Thanks for the fast response!

Is the `0u16` argument the layer_id? Makes sense to keep that.  Also, I'd recommend using an enumeration instead of a boolean for the third 'offset' argument. You could more easily extend it to different options (center, upper left, fixed offset, etc).  I'd also suggest having the reverse mapping of pixel to map (useful for mouse selection).

**Comment by bzm3r** [comment]
Fixed by #268
